arduino-ide-extension/src/node/sketches-service-impl.ts

```diff
@@ -637,6 +628,62 @@ void loop() {
       return false;
     }
   }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private tryParse(raw: string): any | undefined {
+    try {
+      return JSON.parse(raw);
+    } catch {
+      return undefined;
+    }
+  }
+
+  // Returns the default.ino from the settings or from default folder.
+  private async loadDefault(): Promise<string> {
+    const root = await this.root(await this.sketchbookUri());
+    const configDirUri = await this.envVariableServer.getConfigDirUri();
+    const configDirPath = FileUri.fsPath(configDirUri);
+
+    let result: string;
+    result = `void setup() {
+  // put your setup code here, to run once:
+
+}
+      
+void loop() {
+  // put your main code here, to run repeatedly:
+  
+}`;
+
+    try {
+      const raw = await fs.readFile(join(configDirPath, 'settings.json'), {
+        encoding: 'utf8',
+      });
+      const json = this.tryParse(raw);
+      if (json) {
+        const value = json['arduino.sketch.inoBlueprint'];
+
+        const filename =
+          typeof value === 'string' && !!value
+            ? value
+            : `${root}/default/default.ino`;
```

Comment:
This is something IDE2 won't support. It's either given by the user, or the default is used from the code.